lib/interfaces/exports.js

```diff
@@ -19,9 +19,99 @@ var Test = require('../test');
  *
  * @param {Suite} suite Root suite.
  */
-module.exports = function(suite) {
+module.exports = function exportsInterface(suite) {
   var suites = [suite];
 
+  suite.on('pre-require', function(context, file, mocha) {
```

Comment:
I guess `exports` UI assumed `bdd` style already set. So, I copied and pasted from `bdd` UI.

Reply:
I don't understand the changes you applied to lib/interfaces/exports.js. This file is the interface for `ui: exports` and has nothing to do with the fact that mocha.ui is called twice.
The changes to mocha.js are working fine, even when --ui is set twice (mocha.opts and command line).